### ISSUE TITLE
Fix FP S3878: When non-object array is passed to object[] params as first argument

### ISSUE THREAD

**Issue by afpirimoglu** [SonarSource/sonar-dotnet#6894]
### Description

When the first element of the params argument is an array and the array is created on the same statement as the method call being made, S3878 is raised incorrectly.

### Repro steps
```csharp
public void MethodWithObjectParams(params object[] objectArgs)
{
    // Does something
}

// For this following line S3878 is raised correctly:
this.MethodWithObjectParams(new object[] { new int[] { 1, 2, 3, } });
// But for this next line S3878 is raised incorrectly:
this.MethodWithObjectParams(new int[] { 1, 2, 3, } ); // Should not have caused S3878, but it is!

// workaround: do the array creation before the method call
var intArray = new int[] { 1, 2, 3, };
this.MethodWithObjectParams(intArray); // Does not raise S3878, as it shouldn't.
```

### Expected behavior

if the first parameter being passed is an array that is created on the statement call line, but is not the array of the params argument, but an element of the params argument, S3878 should not have been raised.

### Actual behavior

Please see code example above

### Known workarounds

Please see code example above

**Comment by mary-georgiou-sonarsource** [comment]
Hello @afpirimoglu,
Thanks a lot for reporting this, I confirm this is an FP.
I added the issue to our backlog and you can track any progress by following this issue.

**Comment by afpirimoglu** [comment]
@mary-georgiou-sonarsource thanks for looking into this!
I had filed this issue too, which is also a false positive for S3878: https://github.com/SonarSource/sonar-dotnet/issues/6893
There are situations the params arg have to be called by name in the method call, which then requires it to be passed as an array type.
Is it possible to take a look at that issue also?
